include/os/linux/spl/sys/uio.h

```diff
@@ -68,13 +68,15 @@ typedef struct uio {
 	size_t		uio_skip;
 } uio_t;
 
-#define	uio_segflg(uio)			(uio)->uio_segflg
-#define	uio_offset(uio)			(uio)->uio_loffset
-#define	uio_resid(uio)			(uio)->uio_resid
-#define	uio_iovcnt(uio)			(uio)->uio_iovcnt
-#define	uio_iovlen(uio, idx)		(uio)->uio_iov[(idx)].iov_len
-#define	uio_iovbase(uio, idx)		(uio)->uio_iov[(idx)].iov_base
-#define	uio_fault_disable(uio, set)	(uio)->uio_fault_disable = set
+#define	GET_UIO_STRUCT(u)		(u)
```

Comment:
With a little more refactoring it looks like we could avoid the need to define this at all for Linux.  There are only a couple places it's used in common code and really the better thing to do in those case looks like to define a new `zfs_uio_*` platform specific function.

Reply:
I got rid of this define as well as `uio_td()` from the Linux side.